### PR TITLE
Add documentation for odo completion

### PR DESCRIPTION
**Pull request by cdrage**
**What type of PR is this:**

/kind documentation

**What does this PR do / why we need it:**

Adds documentation to command reference with regards to `odo completion`

**Which issue(s) this PR fixes:**

Closes https://github.com/redhat-developer/odo/issues/5536

**PR acceptance criteria:**

- [X] Documentation

**How to test changes / Special notes to the reviewer:**

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: docs/website/docs/command-reference/completion.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: odo completion
+---
+
+## Description
+
+`odo completion` is used to generate shell completion code. The generated code provides interactive shell completion code for `odo`.
+
+There is support for the following terminal shell's:
```
Comment (rm3l) [inline docs/website/docs/command-reference/completion.md:9]:
```suggestion
There is support for the following terminal shells:
```

File: docs/website/docs/command-reference/completion.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+- [Bash](https://www.gnu.org/software/bash/)
+- [Zsh](https://ohmyz.sh/)
```
Comment (rm3l) [inline docs/website/docs/command-reference/completion.md:11]: Surely nitpicking, but I think we should link to the official zsh website instead. People can use other tools.

```suggestion
- [Zsh](https://zsh.sourceforge.io/)
```

**Review by rm3l — CHANGES_REQUESTED** [review]

**Comment by cdrage** [comment]
Thanks @rm3l I've pushed these changes

**Review by rm3l — APPROVED** [review]

**Comment by cdrage** [comment]
> I am okay with the changes, but I was wondering what you think about using tabs instead of headers for the different shells? @cdrage

I'll be changing this in the future once I refactor all the docs, as of right now, all command reference docs don't use tabs as it takes the documentation directly from `--help`.

I'm hoping to create a script to eventually automate all of this.

**Comment by cdrage** [comment]
Going to merge this as I have one LGTM / APPROVE for docs!

**Pull request merged**